File: app/src/main/java/in/completecourse/questionbank/adapter/SpinAdapter.java
package in.completecourse.questionbank.adapter;

import android.content.Context;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import in.completecourse.questionbank.R;

public class SpinAdapter extends ArrayAdapter<String> {
    private final LayoutInflater mLayoutInflater;
    private final String[] mStringArray;
    private final int mResource;

    public SpinAdapter(@NonNull Context context, int resource, String[] strArr) {
        super(context, resource, 0, strArr);
        this.mLayoutInflater = LayoutInflater.from(context);
        this.mResource = resource;
        this.mStringArray = strArr;
    }

    private View homeView(int i, ViewGroup viewGroup){
        View view = this.mLayoutInflater.inflate(this.mResource, viewGroup, false);

        switch (i%3){
            case 0:
            case 2:
                ((TextView)view.findViewById(R.id.offer_type_txt)).setText(Html.fromHtml(this.mStringArray[i] +
                        "<sup><small>th</small></sup>"));
                ((TextView)view.findViewById(R.id.tvclass)).setText("Class");
                break;
            case 1:
                ((TextView)view.findViewById(R.id.offer_type_txt)).setText(Html.fromHtml(this.mStringArray[i] +
                        "<sup><small>वीं</small></sup>"));
                ((TextView)view.findViewById(R.id.tvclass)).setText("कक्षा");
                break;

        }

        return view;
    }

    @Override
    public View getDropDownView(int i, @Nullable View view, @NonNull ViewGroup parent) {
        if (view == null){
            view = this.mLayoutInflater.inflate(R.layout.spinner_new_view, parent, false);
        }

        switch (i%3){
            case 0:
            case 2:
                ((TextView)view.findViewById(R.id.offer_type_txt)).setText(Html.fromHtml(this.mStringArray[i] +
                        "<sup><small>th</small></sup>"));
                ((TextView)view.findViewById(R.id.tvclass)).setText("Class");
                break;
            case 1:
                ((TextView)view.findViewById(R.id.offer_type_txt)).setText(Html.fromHtml(this.mStringArray[i] +
                        "<sup><small>वीं</small></sup>"));
                ((TextView)view.findViewById(R.id.tvclass)).setText("कक्षा");
                break;

        }

        return view;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
           return homeView(position, parent);
    }
}

File: app/src/main/java/in/completecourse/questionbank/adapter/SpinAdapter.kt
package `in`.completecourse.questionbank.adapter

import android.content.Context
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import `in`.completecourse.questionbank.R

class SpinAdapter(context: Context, resource: Int, strArr: Array<String>) : ArrayAdapter<String?>(context, resource, 0, strArr) {
    private val mLayoutInflater: LayoutInflater = LayoutInflater.from(context)
    private val mStringArray: Array<String> = strArr
    private val mResource: Int = resource
    private fun homeView(i: Int, viewGroup: ViewGroup): View {
        val view = mLayoutInflater.inflate(mResource, viewGroup, false)
        when (i % 3) {
            0, 2 -> {
                (view.findViewById<View>(R.id.offer_type_txt) as TextView).text = Html.fromHtml(mStringArray[i] +
                        "<sup><small>th</small></sup>")
                (view.findViewById<View>(R.id.tvclass) as TextView).text = "Class"
            }
            1 -> {
                (view.findViewById<View>(R.id.offer_type_txt) as TextView).text = Html.fromHtml(mStringArray[i] +
                        "<sup><small>वीं</small></sup>")
                (view.findViewById<View>(R.id.tvclass) as TextView).text = "कक्षा"
            }
        }
        return view
    }

    override fun getDropDownView(i: Int, view: View?, parent: ViewGroup): View {
        var myView = view
        if (myView == null) {
            myView = mLayoutInflater.inflate(R.layout.spinner_new_view, parent, false)
        }
        when (i % 3) {
            0, 2 -> {
                (myView!!.findViewById<View>(R.id.offer_type_txt) as TextView).text = Html.fromHtml(mStringArray[i] +
                        "<sup><small>th</small></sup>")
                (myView.findViewById<View>(R.id.tvclass) as TextView).text = "Class"
            }
            1 -> {
                (myView!!.findViewById<View>(R.id.offer_type_txt) as TextView).text = Html.fromHtml(mStringArray[i] +
                        "<sup><small>वीं</small></sup>")
                (myView.findViewById<View>(R.id.tvclass) as TextView).text = "कक्षा"
            }
        }
        return myView!!
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        return homeView(position, parent)
    }

}